gap/SemilinearMatrixGroups.gi

```diff
@@ -194,3 +194,85 @@ function(d, q, s)
     # conjugate the result so that it preserves the standard unitary form 
     return ConjugateToStandardForm(MatrixGroupWithSize(F, generators, size), "U");
 end);
+
+# Construction as in Proposition 6.4 of [HR05]
+BindGlobal("SymplecticSemilinearSp",
+function(d, q, s)
+    local F, gammaL1, As, Bs, m, omega, AandB, C, i, range, generators, size;
+    if d mod s <> 0 or not IsPrime(s) then
+        ErrorNoReturn("<s> must be prime and a divisor of <d> but <s> = ", s,
+                      " and <d> = ", d);
+    fi;
+    if not IsEvenInt(QuoInt(d, s)) then
+        ErrorNoReturn("The quotient <d> / <s> must be even but <d> = ", d, 
+                      " and <s> = ", s);
+    fi;
+    F := GF(q);
+    gammaL1 := MatricesInducingGaloisGroupOfGFQToSOverGFQ(s, q);
+    # Let w be a primitive element of GF(q ^ s) over GF(q). Since As is the
+    # companion matrix of the minimal polynomial of w over GF(q), its
```

Comment:
At this point, `As` is not even defined. Perhaps the line setting `As` should be moved before this comment? Or perhaps say something like
```suggestion
    # Let w be a primitive element of GF(q ^ s) over GF(q). The matrix As we define
    #  next is the companion matrix of the minimal polynomial of w over GF(q), hence its
```